dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/service/FdcIntegrationTest.java

```diff
@@ -28,612 +28,659 @@
+                .traceExecuteFdcGlobalUpdate()
+                .traceAndFilterGetFdcContributions(updatedIds)
+                .traceAndStubSendFdcUpdate(id -> Boolean.TRUE)
+                .traceUpdateFdcs();
+
+        // Call the processDailyFiles() method under test:
+        final var startDate = LocalDate.now();
+        fdcService.processDailyFiles();
+        final var endDate = LocalDate.now();
+
+        final FdcProcessSpy watched = watching.build();
+
+        // Fetch some items of information from the maat-api to use during validation:
+        final var fdcContributions = updatedIds.stream().map(testDataClient::getFdcContribution).toList();
+        final int contributionFileId = watched.getXmlFileResult();
+        final var contributionFile = testDataClient.getContributionFile(contributionFileId);
+
+        softly.assertThat(watched.getGlobalUpdateResponse().isSuccessful()).isTrue(); // 1
+        softly.assertThat(updatedIds).hasSize(3).doesNotContainNull(); // 2.
+        softly.assertThat(watched.getRequestedIds()).containsAll(updatedIds); // 3.
+        softly.assertThat(watched.getSentIds()).containsAll(updatedIds); // 4.
+
+        softly.assertThat(watched.getRecordsSent()).isGreaterThanOrEqualTo(3); // 5.
+        softly.assertThat(watched.getXmlCcIds()).containsAll(updatedIds);
+
+        fdcContributions.forEach(fdcContribution -> { // 6.
+            softly.assertThat(watched.getXmlContent()).contains("<fdc id=\"" + fdcContribution.getId() + "\">");
+            // TODO uncomment the following line - it should work but doesn't at the moment because maat API
+            //      endpoint /debt-collection-enforcement/fdc-contribution returns the maatId as null.
+            // softly.assertThat(watched.getXmlContent()).contains("<maat_id>" + fdcContribution.getMaatId() + "</maat_id>");
+        });
+
+        fdcContributions.forEach(fdcContribution -> { // 7.
+            softly.assertThat(fdcContribution.getStatus()).isEqualTo(FdcContributionsStatus.SENT);
+            softly.assertThat(fdcContribution.getContFileId()).isEqualTo(contributionFileId);
+            softly.assertThat(fdcContribution.getUserModified()).isEqualTo("DCES");
+            softly.assertThat(fdcContribution.getDateModified()).isBetween(startDate, endDate);
+        });
+
+        softly.assertThat(contributionFile.getXmlFileName()).isEqualTo(watched.getXmlFileName()); // 8.
+        softly.assertThat(contributionFile.getId()).isEqualTo(watched.getXmlFileResult());
+        softly.assertThat(contributionFile.getRecordsSent()).isGreaterThanOrEqualTo(3);
+        softly.assertThat(contributionFile.getDateCreated()).isBetween(startDate, endDate);
+        softly.assertThat(contributionFile.getUserCreated()).isEqualTo("DCES");
+        softly.assertThat(contributionFile.getDateModified()).isBetween(startDate, endDate);
+        softly.assertThat(contributionFile.getUserModified()).isEqualTo("DCES");
```

Comment:
These are the uncommented test assertions. Everything else is code re-formatting.